core/src/main/java/software/amazon/awssdk/core/async/SingleByteArrayAsyncRequestProvider.java

```diff
@@ -34,6 +34,15 @@ public long contentLength() {
 
     @Override
     public void subscribe(Subscriber<? super ByteBuffer> subscriber) {
+        // FIXME missing protection abiding to rule 1.9, proposal:
+        // As per rule 1.09, we need to throw a `java.lang.NullPointerException`
+        // if the `Subscriber` is `null`
+        // if (subscriber == null) throw null;
+
+        // FIXME: onSubscribe is user code, and could be ill behaved, as library we should protect from this,
+        // FIXME: This is covered by spec rule 2.13; proposal:
+        // As per 2.13, this method must return normally (i.e. not throw).
+        // try {
```

Comment:
guards such as this are required by the spec, they're not hard to add. The TCK will fail and show which rule has been broken in the existing implemenations pointing these out.